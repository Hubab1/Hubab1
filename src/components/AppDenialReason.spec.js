import React from 'react';
import { shallow } from 'enzyme';

import DenialReason from 'components/AppDenialReason';


let defaultProps;

beforeEach(() => {
    defaultProps = {
        date: '5/6/2020',
        buildingName: 'Monterey Pines Apartments',
        unitNumber: 'Unit 14F',
        name: 'John Doe',
        onAgree: jest.fn(),
    };
});

it('Matches Snapshot', function() {
    let wrapper = shallow( <DenialReason {...defaultProps}/> );
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Do not display adverseFactorsList when no factor', function() {
    let wrapper = shallow( <DenialReason {...defaultProps} adverseFactors={[]}/> );
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Display adverseFactorsList when Factors', function() {
    const factors = [
        'Too few open revolving accounts',
        'Bankcard account balances are too high in proportion to credit limits',
        'Insufficient payment activity',
        'Not enough debt experience',
    ];
    let wrapper = shallow( <DenialReason {...defaultProps} adverseFactors={factors}/> );
    expect(wrapper.getElement()).toMatchSnapshot();
});