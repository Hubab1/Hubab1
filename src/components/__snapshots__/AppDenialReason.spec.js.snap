// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display adverseFactorsList when Factors 1`] = `
<div>
  <ForwardRef(render)>
    <ForwardRef(render)>
      Reason for Our Decision
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Adverse action notice
    </ForwardRef(render)>
  </ForwardRef(render)>
  <ForwardRef(render)
    id="denialDecisionReason"
  >
    <ForwardRef(render)
      color="#818797"
    >
      5/6/2020
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Dear 
      John Doe
      ,
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Thank you for your recent rental application for 
      Monterey Pines Apartments
      <React.Fragment>
        ,
      </React.Fragment>
      Unit 14F
      .
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Unfortunately, we are unable to approve your rental application.
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)
      id="adverseFactorsList"
    >
      <ForwardRef(render)>
        <ForwardRef(render)>
          <ForwardRef(render)>
            Factors affecting our decision:
          </ForwardRef(render)>
        </ForwardRef(render)>
        <ForwardRef(render)>
          <ForwardRef(render)>
            <li>
              Too few open revolving accounts
            </li>
            <li>
              Bankcard account balances are too high in proportion to credit limits
            </li>
            <li>
              Insufficient payment activity
            </li>
            <li>
              Not enough debt experience
            </li>
          </ForwardRef(render)>
        </ForwardRef(render)>
      </ForwardRef(render)>
    </ForwardRef(render)>
    <ForwardRef(render)>
      In evaluating your application, 
      <b>
        TransUnion Consumer Solutions
      </b>
       provided us with information that in whole or in part influenced our decision. 
      <b>
        The reporting agency played no part in our decision other than providing us with information about you.
      </b>
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Under the Fair Credit Reporting Act, you have the right to know the information provided to us, which can be obtained by contacting:
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      <b>
        TransUnion Consumer Solutions
      </b>
      <br />
      P.O. Box 2000
      <br />
      Chester, PA. 19022-2000
      <br />
      <a
        href="tel:1-800-916-8800"
      >
        1-800-916-8800
      </a>
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Background Data Solutions
      <br />
      <a
        href="tel:1-800-568-5665"
      >
        1-800-568-5665
      </a>
      <br />
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      You have the right to a free copy of your report from the reporting agency, if you request it no later than 60 days after you receive this notice. In addition, if you find that any information contained in the report you receive is inaccurate or incomplete, you have the right to dispute the matter with the reporting agency that provided the report or through 
      <b>
        TransUnion Rental Screening Solutions
      </b>
       consumer relations at 
      <a
        href="tel:1-800-230-9376"
      >
        1-800-230-9376
      </a>
       or 
      <a
        href="mailto:consumer_relations@turss.com"
      >
        consumer_relations@turss.com
      </a>
      .
    </ForwardRef(render)>
    <br />
  </ForwardRef(render)>
  <ActionButton
    color="primary"
    marginTop={39}
    onClick={[MockFunction]}
    successGreen={false}
    variant="contained"
  >
    OK
  </ActionButton>
</div>
`;

exports[`Do not display adverseFactorsList when no factor 1`] = `
<div>
  <ForwardRef(render)>
    <ForwardRef(render)>
      Reason for Our Decision
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Adverse action notice
    </ForwardRef(render)>
  </ForwardRef(render)>
  <ForwardRef(render)
    id="denialDecisionReason"
  >
    <ForwardRef(render)
      color="#818797"
    >
      5/6/2020
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Dear 
      John Doe
      ,
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Thank you for your recent rental application for 
      Monterey Pines Apartments
      <React.Fragment>
        ,
      </React.Fragment>
      Unit 14F
      .
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Unfortunately, we are unable to approve your rental application.
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      In evaluating your application, 
      <b>
        TransUnion Consumer Solutions
      </b>
       provided us with information that in whole or in part influenced our decision. 
      <b>
        The reporting agency played no part in our decision other than providing us with information about you.
      </b>
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Under the Fair Credit Reporting Act, you have the right to know the information provided to us, which can be obtained by contacting:
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      <b>
        TransUnion Consumer Solutions
      </b>
      <br />
      P.O. Box 2000
      <br />
      Chester, PA. 19022-2000
      <br />
      <a
        href="tel:1-800-916-8800"
      >
        1-800-916-8800
      </a>
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Background Data Solutions
      <br />
      <a
        href="tel:1-800-568-5665"
      >
        1-800-568-5665
      </a>
      <br />
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      You have the right to a free copy of your report from the reporting agency, if you request it no later than 60 days after you receive this notice. In addition, if you find that any information contained in the report you receive is inaccurate or incomplete, you have the right to dispute the matter with the reporting agency that provided the report or through 
      <b>
        TransUnion Rental Screening Solutions
      </b>
       consumer relations at 
      <a
        href="tel:1-800-230-9376"
      >
        1-800-230-9376
      </a>
       or 
      <a
        href="mailto:consumer_relations@turss.com"
      >
        consumer_relations@turss.com
      </a>
      .
    </ForwardRef(render)>
    <br />
  </ForwardRef(render)>
  <ActionButton
    color="primary"
    marginTop={39}
    onClick={[MockFunction]}
    successGreen={false}
    variant="contained"
  >
    OK
  </ActionButton>
</div>
`;

exports[`Matches Snapshot 1`] = `
<div>
  <ForwardRef(render)>
    <ForwardRef(render)>
      Reason for Our Decision
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Adverse action notice
    </ForwardRef(render)>
  </ForwardRef(render)>
  <ForwardRef(render)
    id="denialDecisionReason"
  >
    <ForwardRef(render)
      color="#818797"
    >
      5/6/2020
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Dear 
      John Doe
      ,
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Thank you for your recent rental application for 
      Monterey Pines Apartments
      <React.Fragment>
        ,
      </React.Fragment>
      Unit 14F
      .
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Unfortunately, we are unable to approve your rental application.
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      In evaluating your application, 
      <b>
        TransUnion Consumer Solutions
      </b>
       provided us with information that in whole or in part influenced our decision. 
      <b>
        The reporting agency played no part in our decision other than providing us with information about you.
      </b>
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Under the Fair Credit Reporting Act, you have the right to know the information provided to us, which can be obtained by contacting:
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      <b>
        TransUnion Consumer Solutions
      </b>
      <br />
      P.O. Box 2000
      <br />
      Chester, PA. 19022-2000
      <br />
      <a
        href="tel:1-800-916-8800"
      >
        1-800-916-8800
      </a>
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      Background Data Solutions
      <br />
      <a
        href="tel:1-800-568-5665"
      >
        1-800-568-5665
      </a>
      <br />
    </ForwardRef(render)>
    <br />
    <ForwardRef(render)>
      You have the right to a free copy of your report from the reporting agency, if you request it no later than 60 days after you receive this notice. In addition, if you find that any information contained in the report you receive is inaccurate or incomplete, you have the right to dispute the matter with the reporting agency that provided the report or through 
      <b>
        TransUnion Rental Screening Solutions
      </b>
       consumer relations at 
      <a
        href="tel:1-800-230-9376"
      >
        1-800-230-9376
      </a>
       or 
      <a
        href="mailto:consumer_relations@turss.com"
      >
        consumer_relations@turss.com
      </a>
      .
    </ForwardRef(render)>
    <br />
  </ForwardRef(render)>
  <ActionButton
    color="primary"
    marginTop={39}
    onClick={[MockFunction]}
    successGreen={false}
    variant="contained"
  >
    OK
  </ActionButton>
</div>
`;
